File: Restaurant/Restaurant.Web/Modules/ServerTypes/Default/BrandColumns.ts
import { ImageViewerFormatter } from "@/Default/Product/ImageViewerFormatter";
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BrandRow } from "./BrandRow";

export interface BrandColumns {
    Id: Column<BrandRow>;
    BrandName: Column<BrandRow>;
    Photo: Column<BrandRow>;
    PhoneNember: Column<BrandRow>;
    WhtasApp: Column<BrandRow>;
    FaceBook: Column<BrandRow>;
    Twitter: Column<BrandRow>;
    Instagram: Column<BrandRow>;
    Tiktok: Column<BrandRow>;
    Telegram: Column<BrandRow>;
    Description: Column<BrandRow>;
}

export class BrandColumns extends ColumnsBase<BrandRow> {
    static readonly columnsKey = 'Default.Brand';
    static readonly Fields = fieldsProxy<BrandColumns>();
}

[ImageViewerFormatter]; // referenced types

File: Restaurant/Restaurant.Web/Modules/Default/Brand/BrandColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Restaurant.Default.Columns;

[ColumnsScript("Default.Brand")]
[BasedOnRow(typeof(BrandRow), CheckNames = true)]
public class BrandColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId")]
    public int Id { get; set; }
    [EditLink]
    public string BrandName { get; set; }
    [ImageViewerFormatter]
    public string Photo { get; set; }
    public string PhoneNember { get; set; }
    public string WhtasApp { get; set; }
    public string FaceBook { get; set; }
    public string Twitter { get; set; }
    public string Instagram { get; set; }
    public string Tiktok { get; set; }
    public string Telegram { get; set; }
    public string Description { get; set; }
}